### PR TITLE
Add some notes about perfetto in DEVDOC

### PR DESCRIPTION
**Pull request by hevrard**

### CODE REVIEW

File: DEVDOC.md [hunk 1]
```diff
@@ -685,3 +685,34 @@ only the (transformed) capture commands needs to be replayed.
...
+
+## Life of a perfetto trace (system profile trace)
+
+AGI heavily relies on the [perfetto](https://perfetto.dev/) system profiling
+framework to obtain, store and process profiling data. Hence we often refer to
+system profile traces as "perfetto traces". If you have to deal with perfetto
+code, make sure to refer to the [perfetto
+documentation](https://perfetto.dev/docs/).
+
+Perfetto is build into Android (since Android 9 Pie). In general, to take a
```
Comment (purvisa-at-google-com) [inline DEVDOC.md:697]: build -> built

File: DEVDOC.md [hunk 2]
```diff
@@ -685,3 +685,34 @@ only the (transformed) capture commands needs to be replayed.
...
+perfetto trace, you can use the `perfetto` command-line tool on the device. See
+perfetto's web interface at https://ui.perfetto.dev/ and click on "Recording
+command" to see an example of how the `perfetto` command-line tool can be used
+to obtain a profiling trace.
+
+AGI does not use the `perfetto` command line tool. It directly interacts with
+perfetto's `traced` deamon running on the device. This deamon listens to the
+`/dev/socket/traced_consumer`, and AGI connects directly to this socket. The
+related AGI code is under `gapis/perfetto/client/`.
```
Comment (purvisa-at-google-com) [inline DEVDOC.md:706]: There is another code path that uses the CLI to somehow activate the trace by catting things into files as far as I know.

**Review by purvisa-at-google-com — APPROVED** [review]
Looks good generally!

File: DEVDOC.md (hunk 1 above)
Comment (hevrard) [reply DEVDOC.md:697]: done

File: DEVDOC.md (hunk 2 above)
Comment (hevrard) [reply DEVDOC.md]: Thanks for this, I found it! I clarified the text accordingly.

**Pull request merged**